src/NUnitFramework/framework/Internal/ValueTaskAwaitAdapter.cs

```diff
@@ -11,21 +10,21 @@ internal static class ValueTaskAwaitAdapter
     {
         public static AwaitAdapter Create(ValueTask task)
         {
-            var genericValueTaskType = task
-                .GetType()
-                .TypeAndBaseTypes()
```

Comment:
Why is it that this is no longer needed ?

Reply:
This was originally taken from the Task await adapter, where for the generic task this method would return [Task<>, Task, object]. This was probably to get Task<> and Task in the same sequence, though in my opinion it was a bit excessive to begin with.

For ValueTask however, it doesn’t work to begin with, because ValueTask<> does not inherit from ValueTask, since they’re structs. So the handling has to be separate.